src/legacy/ui/public/timefilter/timefilter.ts

```diff
@@ -19,67 +19,77 @@
 
 import _ from 'lodash';
 import { Subject, BehaviorSubject } from 'rxjs';
-import moment from 'moment';
-import { calculateBounds, getTime } from './get_time';
-import { parseQueryString } from './lib/parse_querystring';
+import moment, { Moment } from 'moment';
 import { subscribeWithScope } from 'ui/utils/subscribe_with_scope';
-import uiRoutes from '../routes';
 import chrome from 'ui/chrome';
-import { areTimePickerValsDifferent } from './lib/diff_time_picker_vals';
+import { UiSettingsClientContract } from 'src/core/public';
+import { RefreshInterval, TimeRange } from 'src/plugins/data/public';
+import { IndexPattern } from 'src/legacy/core_plugins/data/public';
+import { IScope } from 'angular';
 import { timeHistory } from './time_history';
+import { areRefreshIntervalsDifferent, areTimeRangesDifferent } from './lib/diff_time_picker_vals';
+import uiRoutes from '../routes';
+import { parseQueryString } from './lib/parse_querystring';
+import { calculateBounds, getTime } from './get_time';
 
-class Timefilter {
-  constructor() {
-
-    // Fired when isTimeRangeSelectorEnabled \ isAutoRefreshSelectorEnabled are toggled
-    this.enabledUpdated$ = new BehaviorSubject();
-
-    // Fired when a user changes the timerange
-    this.timeUpdate$ = new Subject();
-
-    // Fired when a user changes the the autorefresh settings
-    this.refreshIntervalUpdate$ = new Subject();
-
-    // Used when search poll triggers an auto refresh
-    this.autoRefreshFetch$ = new Subject();
-
-    this.fetch$ = new Subject();
+// Timefilter accepts moment input but always returns string output
+export type InputTimeRange =
+  | TimeRange
+  | {
+      from: Moment;
+      to: Moment;
+    };
 
-    this.isTimeRangeSelectorEnabled = false;
-    this.isAutoRefreshSelectorEnabled = false;
-    this._time = chrome.getUiSettingsClient().get('timepicker:timeDefaults');
-    this.setRefreshInterval(chrome.getUiSettingsClient().get('timepicker:refreshIntervalDefaults'));
+export class Timefilter {
+  // Fired when isTimeRangeSelectorEnabled \ isAutoRefreshSelectorEnabled are toggled
+  private enabledUpdated$ = new BehaviorSubject(false);
+  // Fired when a user changes the timerange
+  private timeUpdate$ = new Subject();
+  // Fired when a user changes the the autorefresh settings
+  private refreshIntervalUpdate$ = new Subject();
+  // Used when search poll triggers an auto refresh
+  private autoRefreshFetch$ = new Subject();
+  private fetch$ = new Subject();
+
+  private _time: TimeRange;
+  private _refreshInterval!: RefreshInterval;
+
+  public isTimeRangeSelectorEnabled: boolean = false;
+  public isAutoRefreshSelectorEnabled: boolean = false;
+
+  constructor(uiSettings: UiSettingsClientContract) {
+    this._time = uiSettings.get('timepicker:timeDefaults');
```

Comment:
should we rather pass this settings directly to constructor, so Timefilter has no dependency on uiSettingsClient ?

Reply:
YES. I will clean it up when moving to the `data` plugin.